app/src/main/java/org/simple/clinic/analytics/FirebaseAnalyticsReporter.kt

```diff
@@ -0,0 +1,40 @@
+package org.simple.clinic.analytics
+
+import android.app.Application
+import androidx.core.os.bundleOf
+import com.google.firebase.analytics.FirebaseAnalytics
+import org.simple.clinic.platform.analytics.AnalyticsReporter
+import org.simple.clinic.platform.analytics.AnalyticsUser
+import org.simple.clinic.util.unsafeLazy
+
+class FirebaseAnalyticsReporter(
+    private val application: Application
+) : AnalyticsReporter {
+
+  private val firebaseAnalytics by unsafeLazy {
+    FirebaseAnalytics.getInstance(application)
+  }
+
+  override fun setLoggedInUser(user: AnalyticsUser, isANewRegistration: Boolean) {
+    synchronized(firebaseAnalytics) {
+      val userId = user.id.toString()
+
+      // The current facility, deployment, etc are being tracked in `FirebaseAnalyticsInfrastructure`
+      firebaseAnalytics.setUserId(userId)
+      firebaseAnalytics.setUserProperty("is_new_registration", isANewRegistration.toString())
```

Comment:
Why are we setting additional user property `is_new_registration`? It was not present in the mixpanel reporter.

Reply:
Mixpanel had a way to identify new registrations. Since Firebase doesn't have it natively, we are just using a property to identify new registrations.